src/agent/srcview/src/pdbcache.rs

```diff
@@ -47,57 +133,74 @@ impl PdbCache {
+                let fname = match id_finder.find(inlinee.index())?.parse()? {
+                    pdb::IdData::Function(fid) => fid.name,
+                    pdb::IdData::MemberFunction(mid) => mid.name,
+                    _ => {
+                        bail!("Expected LF_MFUC_ID or LF_FUNC_ID for {}", inlinee.index());
+                    }
+                };
+                inlinees.insert(inlinee.index(), (inlinee, fname));
+            }
+
             let program = info.line_program()?;
             let mut symbols = info.symbols()?;
 
             while let Some(symbol) = symbols.next()? {
-                if let Ok(SymbolData::Procedure(proc)) = symbol.parse() {
-                    let proc_name = proc.name.to_string();
-                    let mut lines = program.lines_for_symbol(proc.offset);
-
-                    let symbol_to_lines = symbol_to_lines.entry(proc_name.to_string()).or_default();
-
-                    while let Some(line_info) = lines.next()? {
-                        let rva = line_info
-                            .offset
-                            .to_rva(&address_map)
-                            .ok_or_else(|| format_err!("invalid RVA: {:?}", line_info))?;
-                        let file_info = program.get_file_info(line_info.file_index)?;
-                        let file_name = file_info.name.to_string_lossy(&string_table)?;
-
-                        let path = file_name.into_owned();
-                        let line = line_info.line_start as usize;
-
-                        let srcloc = SrcLine::new(path.clone(), line);
-
-                        offset_to_line.insert(rva.0 as usize, srcloc.clone());
-                        path_to_symbols
-                            .entry(path.clone())
-                            .or_default()
-                            .push(proc_name.to_string());
-                        symbol_to_lines.push(srcloc.clone());
-                        path_to_lines.entry(path.clone()).or_default().push(line);
+                match symbol.parse() {
+                    Ok(SymbolData::Procedure(proc)) => {
+                        let lines = program.lines_for_symbol(proc.offset);
+                        let proc_name = proc.name.to_string();
+                        builder.update_from_iter(&program, &string_table, &address_map, &proc_name, lines)?;
+                    }
+                    Ok(SymbolData::InlineSite(site)) => {
+                        // Locate the parent proc of this inline site
+                        let mut sid = site.parent;
+                        let offset;
+                        loop {
+                            let sid_actual = sid.ok_or_else(|| format_err!("S_INLINESITE should have parent"))?;
+                            let mut tmp_iter = info.symbols_at(sid_actual)?;
+                            let data = tmp_iter.next()?
+                                .ok_or_else(|| format_err!("Unable to parse symbol at {sid_actual}"))?
```

Comment:
Seems like this error message should be `.context` on the `parse` result? If `tmp_iter` is empty it should be something like "symbol not found".

Reply:
`tmp_iter.next()?` is an `Option`. Trying to just convert it to a `Result`, but I think I might clarify the error message to specify that the symbol cannot be found